src/simple_experiments/events/simple_card.cljs

```diff
@@ -25,23 +19,63 @@
     :pending-registration
     :associated})
 
-(defn set-active-card [{:keys [db]} [_ card-uuid status]]
-  (if (nil? card-uuid)
-    {}
-    (let [sdid (six-digit-id card-uuid)]
-      {:db (assoc-in db [:ui :active-card]
-                     {:uuid card-uuid
-                      :six-digit-id sdid
-                      :six-digit-display (six-digit-display sdid)
-                      :status status})})))
+(defn has-six-digit-id? [patient six-digit-id]
+  (let [six-digit-ids (set (concat (:six-digit-ids patient)
+                                   (map ->six-digit-id
+                                        (:card-uuids patient))))]
+    (contains? six-digit-ids six-digit-id)))
+
+(defn find-patients [db six-digit-id]
+  (->> db
+       :store
+       :patients
+       vals
+       (filter #(has-six-digit-id? % six-digit-id))))
+
+(defn handle-six-digit-keyboard [{:keys [db]} [_ six-digit-id]]
```

Comment:
I read this and assumed the function had something to do with the android `keyboard` :). Which, doesn't appear to be the case. Have you seen `assoc-into-db` in `events`?

Reply:
1. We are already following this convention:
```clojure
(defn handle-bp-keyboard [{:keys [db]} [_ kind value]]
  (cond
    (and (= kind :systolic)
         (or (and (re-find #"^[12]" value)
                  (= 3 (count value)))
             (and (not (re-find #"^[12]" value))
                  (= 2 (count value)))))
    (.focus (get-in db [:ui :bp :diastolic :ref])))
  {:db (assoc-in db [:ui :bp :value kind] value)})
```
2. `assoc-into-db` returns a `fn`

Reply:
We do this when we're writing a full handler anyway. `assoc-into-db` is not meant to be composed. It's meant to be a simple replacement for the full event handler.